pkg/proxy/proxy.go

```diff
@@ -90,3 +83,36 @@ func appendClientCertAuthHeaders(header http.Header, user userinfo.Info, UserHea
 		header.Add(GroupHeader, group)
 	}
 }
+
+func newShardReverseProxy() *httputil.ReverseProxy {
+	director := func(req *http.Request) {
+		shardURL := ShardURLFrom(req.Context())
+		if shardURL == nil {
+			// should not happen if wiring is correct
+			runtime.HandleError(fmt.Errorf("no shard URL found in request context"))
+			req.URL.Scheme = "https"
+			req.URL.Host = "notfound"
+			return
+		}
+
+		req.URL.Scheme = shardURL.Scheme
+		req.URL.Host = shardURL.Host
+	}
+	return &httputil.ReverseProxy{Director: director}
```

Comment:
why not `NewSingleHostReverseProxy` ?

Reply:
does too much magic, e.g. wipes the user-agent.